lib/reservations.rb

```diff
@@ -0,0 +1,51 @@
+require 'date'
+require_relative 'hotel.rb'
+require_relative 'room.rb'
+
+
+module Hotel_System
+
+  class Reservations
+    attr_accessor :all_rooms, :check_in, :check_out, :room, :today, :num_of_nights, :all_reservations, :total_cost, :date_range
+
+    def initialize(room, check_in, check_out)
+      @room = room
+      @check_in = Date.parse(check_in)
+      @check_out = Date.parse(check_out)
+      @today = Date.today
+    end
+
+    def valid_date_range?
```

Comment:
`initialize` should probably call this method intrinsically.  You don't want it possible to create a Reservation without valid dates.